source/MRTest/MRPointCloudVariadicOffsetTests.cpp

```diff
@@ -95,4 +100,68 @@ TEST( MRMesh, findClosestWeightedMeshPoint )
     }
 }
 
+TEST( MRMesh, weightedMeshShell )
+{
+    auto cube = makeCube();
+    auto rot = Matrix3f::rotation( { 1.f, 1.f, 1.f }, PI_F / 6.f );
+    for ( auto& pt : cube.points )
+        pt = rot * pt;
+
+    auto offCube = weightedMeshShell( cube, WeightedPointsShellParametersRegions{ { 0.02f, 0.01f, 10.f }, {}, 0.f, false } );
```

Comment:
Is 10 extra layers important for the test?

Reply:
Potentially, it highlights more errors.